### PR TITLE
Replacing sklearn.cross_validation with sklearn.model_selection

### PR DESCRIPTION
**Pull request by kchawla-pi**
Fix for issue #1631

### CODE REVIEW

**Comment by kchawla-pi** [comment]
The Travis build failed on one of the three Python2.7 builds. The failing build seems to have Sci-kit Learn v15.0, while the build passed on SKL v 15.1 . 
How should this be handled? Do we do a version check & run the older or newer example file accordingly?

**Comment by kchawla-pi** [comment]
The message is
[error] 0.66% nilearn.datasets.tests.test_atlas.test_fetch_atlas_pauli_2017: 1.8549s
This was passing earlier and I haven't touched it. Rerunning it and checking.

**Comment by kchawla-pi** [comment]
TravisCI error disappeared spontaneously.
I changed one example file & validated it. However, CircleCI Builds are not passing, seemingly because `brainomics localizer` database is down.
I don't wanna keep making changes if I can't test them here as I go. 
Suggestions?

**Comment by KamalakerDadi** [comment]
> Suggestions?

You can test the changes locally with the pre-downloaded data from brainomics.

**Comment by kchawla-pi** [comment]
@KamalakerDadi That worked. Thanks.
Somehow, CircleCI on the server also passed.

**Comment by kchawla-pi** [comment]
I had halted making changes when this kept failing, wondering what mistake I might have made. I can now proceed.

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 1]
```diff
@@ -79,17 +79,19 @@
...
-
-        mask_chance_scores[mask_name][category] = cross_val_score(
-            dummy_classifier,
-            masked_timecourses,
-            classification_target,
-            cv=cv, scoring="roc_auc")
+        mask_scores[mask_name][category] = cross_val_score(classifier,
+                                                           masked_timecourses,
+                                                           classification_target,
+                                                           cv=cv.split(X=masked_timecourses, groups=session_labels),
```
Comment (KamalakerDadi) [inline examples/02_decoding/plot_haxby_full_analysis.py]: PEP8. Too long lines to read.
https://www.python.org/dev/peps/pep-0008/?#maximum-line-length

I think you should adapt to like what it was before.

File: examples/plot_decoding_tutorial.py [hunk 2]
```diff
@@ -215,13 +215,13 @@
...
+# and faces. To leave a session out, we pass it to the split method of a
+# LeaveOneGroupOut object:
 session_label = behavioral['chunks'][condition_mask]
 
-from sklearn.cross_validation import LeaveOneLabelOut
-cv = LeaveOneLabelOut(session_label)
-cv_score = cross_val_score(svc, fmri_masked, conditions, cv=cv)
+from sklearn.model_selection import LeaveOneGroupOut
+cv = LeaveOneGroupOut()
+cv_score = cross_val_score(svc, fmri_masked, conditions, cv=cv.split(X=fmri_masked, groups=session_label))
```
Comment (KamalakerDadi) [inline examples/plot_decoding_tutorial.py]: same here. lines too long to read.

File: examples/02_decoding/plot_haxby_full_analysis.py (hunk 1 above)
Comment (kchawla-pi) [reply examples/02_decoding/plot_haxby_full_analysis.py]: I find code clarity suffers when the arguments are indented so far from the function names, which is why I don't do that.
PEP8 suggests capping at 99 chars length instead of 79 if the long sentences are preferred, so I will shorten these to 99 chars length.
The 79 char convention was necessary when the screens were lo-res 4:3. 
FHD 16:9 & 3:2 should be comfortable with 99 chars length.

Comment (KamalakerDadi) [reply examples/02_decoding/plot_haxby_full_analysis.py]: Whatever you do flake8 should be happy given that we have one such travis environment. Normally, people don't care about this. But, I will care.

**Comment by kchawla-pi** [comment]
Have replaced cross-validation with mode-selection in all the examples except the one blocked by SearchLight issue #1694 , present in 2 examples.
After address issue #1694 , will make the changes there as well.
After that, will make the replacements in the 4 non-example locations.
```
nilearn/_utils/fixes/__init__.py  # 3 usages
nilearn/decoding/tests/test_searchlight.py  # 1 usage
```

**Comment by kchawla-pi** [comment]
I have modified some of the comments/ explanations in the examples as well, not all. Whoever reviews this, please check that my changes to the comments are appropriate, and point out any where I haven't changed and should.

**Comment by kchawla-pi** [comment]
Would it be better to merge this so most of the examples are updated and then a new PR for subsequent updates or to wait until they are all in? What would be the pros & cons of both approaches?

File: doc/themes/nilearn/layout.html [hunk 3]
```diff
@@ -105,10 +105,10 @@
         for(i in sections){
             if(sections[i] > pos){
 		break;
-            };
-	    if($('a.internal[href$="' + i + '"]').is(':visible')){
+            }
+            if($('a.internal[href$="' + i + '"]').is(':visible')){
```
Comment (bthirion) [inline doc/themes/nilearn/layout.html]: This seems unrelated.
Pelase try to keep PRs specific.

File: doc/themes/nilearn/static/copybutton.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ $(document).ready(function() {
     var div = $('.highlight-python .highlight,' +
                 '.highlight-python3 .highlight,' + 
                 '.highlight-pycon .highlight,' +
-		'.highlight-default .highlight')
+		'.highlight-default .highlight');
```
Comment (bthirion) [inline doc/themes/nilearn/static/copybutton.js]: same here.

File: doc/themes/nilearn/static/copybutton.js [hunk 5]
```diff
@@ -21,14 +21,14 @@ $(document).ready(function() {
...
-    }
+    };
 
     // create and add the button to all the code blocks that contain >>>
     div.each(function(index) {
         var jthis = $(this);
         if (jthis.find('.gp').length > 0) {
             var button = $('<span class="copybutton">&gt;&gt;&gt;</span>');
-            button.css(button_styles)
+            button.css(button_styles);
```
Comment (bthirion) [inline doc/themes/nilearn/static/copybutton.js]: and here.

File: doc/themes/nilearn/layout.html (hunk 3 above)
Comment (kchawla-pi) [reply doc/themes/nilearn/layout.html]: I don't know why the HTML & JS files changed. I used Sphinx to build them for testing. I didn't change their code manually. Seems they were inadvertently auto-added and committed by Git.

File: doc/themes/nilearn/static/copybutton.js (hunk 4 above)
Comment (kchawla-pi) [reply doc/themes/nilearn/static/copybutton.js]: Same as above.

File: doc/themes/nilearn/static/copybutton.js (hunk 5 above)
Comment (kchawla-pi) [reply doc/themes/nilearn/static/copybutton.js]: Same as above.

**Comment by kchawla-pi** [comment]
I think I should do a squash first, for a neater commit history.
Time for an interactive rebase I think.
I have butchered rebases in the past, will do it under guidance.

**Comment by GaelVaroquaux** [comment]
> I think I should do a squash first, for a neater commit history.

I would not worry about it: it's just cosmetic for the commit history. No
need to loose time on it.

File: examples/02_decoding/plot_haxby_anova_svm.py [hunk 6]
```diff
@@ -88,15 +88,15 @@
...
 # Define the cross-validation scheme used for validation.
-# Here we use a LeaveOneLabelOut cross-validation on the session label
+# Here we use a LeaveOneGroupOut cross-validation on the session group
 # which corresponds to a leave-one-session-out
-cv = LeaveOneLabelOut(session)
+cv = LeaveOneGroupOut()
 
 # Compute the prediction accuracy for the different folds (i.e. session)
-cv_scores = cross_val_score(anova_svc, X, conditions, cv=cv)
+cv_scores = cross_val_score(anova_svc, X, conditions, cv=cv.split(X=X, groups=session))
```
Comment (GaelVaroquaux) [inline examples/02_decoding/plot_haxby_anova_svm.py]: This is not the right fix: you need to pass "groups=session" to cross_val_score and leave cv unchanged.

File: examples/02_decoding/plot_haxby_different_estimators.py [hunk 7]
```diff
@@ -114,17 +114,26 @@
...
-            classifiers_scores[classifier_name][category].mean(),
-            classifiers_scores[classifier_name][category].std(),
-            time.time() - t0))
+        classifiers_scores[classifier_name][category] = cross_val_score(classifier,
+                                                                        masked_timecourses,
+                                                                        classification_target,
+                                                                        cv=cv.split(
+                                                                                X=masked_timecourses,
+                                                                                groups=session_labels
+                                                                                ),
```
Comment (GaelVaroquaux) [inline examples/02_decoding/plot_haxby_different_estimators.py]: Same comment: you need to pass groups=session_label to cross_val_score.

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 8]
```diff
@@ -79,17 +79,25 @@
...
-            masked_timecourses,
-            classification_target,
-            cv=cv, scoring="roc_auc")
+        mask_scores[mask_name][category] = cross_val_score(classifier,
+                                                           masked_timecourses,
+                                                           classification_target,
+                                                           cv=cv.split(
+                                                                   X=masked_timecourses,
+                                                                   groups=session_labels,
+                                                                   ),
```
Comment (GaelVaroquaux) [inline examples/02_decoding/plot_haxby_full_analysis.py]: Same thing here.

File: examples/02_decoding/plot_haxby_anova_svm.py (hunk 6 above)
Comment (kchawla-pi) [reply examples/02_decoding/plot_haxby_anova_svm.py]: Okay, I'll check that out. That would also obviate the SearchLight issue that I filed.

File: examples/03_connectivity/plot_group_level_connectivity.py [hunk 9]
```diff
@@ -182,22 +182,33 @@ def plot_matrices(matrices, matrix_kind):
...
     svc = LinearSVC(random_state=0)
-    cv_scores = cross_val_score(svc, connectivity_biomarkers[kind],
-                                y=adhd_labels, cv=cv, scoring='accuracy')
+    cv_scores = cross_val_score(svc,
+                                connectivity_biomarkers[kind],
+                                y=adhd_labels,
+                                cv=cv.split(
+                                        X=np.zeros(len(adhd_labels)),  # dummy data
+                                        y=adhd_labels,
+                                        ),
```
Comment (GaelVaroquaux) [inline examples/03_connectivity/plot_group_level_connectivity.py]: Same problem here

File: examples/plot_decoding_tutorial.py [hunk 10]
```diff
@@ -215,13 +218,17 @@
...
 
-from sklearn.cross_validation import LeaveOneLabelOut
-cv = LeaveOneLabelOut(session_label)
-cv_score = cross_val_score(svc, fmri_masked, conditions, cv=cv)
+from sklearn.model_selection import LeaveOneGroupOut
+cv = LeaveOneGroupOut()
+cv_score = cross_val_score(svc,
+                           fmri_masked,
+                           conditions,
+                           cv=cv.split(X=fmri_masked, groups=session_label),
```
Comment (GaelVaroquaux) [inline examples/plot_decoding_tutorial.py]: Here too :)

**Comment by GaelVaroquaux** [comment]
OK, no further comment :).

The big deal is really deciding what we do with tension between supporting old version of sklearn and new one. We probably will have to drop the old ones, and too bad for people stuck on old versions.

**Comment by bthirion** [comment]
+1 for dropping old version. 
Should we set the default sklearn to 0.18 ? 
B

**Comment by kchawla-pi** [comment]
+1 for Yay to the new and nay to the old version.

**Comment by GaelVaroquaux** [comment]
> +1 for dropping old version.
> Should we set the default sklearn to 0.18 ?

Sounds good.

Kshitij, can you do it in this PR? It requires bumping the version in
nilearn/version, updating the docs, and the CI.

If we're doing this, we might as well bump numpy to 
1.9, and scipy to 0.15.

Bumping these version will allow us to remove nilearn/_utils/fixes.py and
clean up the nilearn code that is currently using it.

**Comment by kchawla-pi** [comment]
scikit-learn==0.18 needs numpy==1.11 . Will update after testing locally.

**Comment by kchawla-pi** [comment]
Gael & Jerome were facing a problem on July2-3. Increasing the minimum version numbers for packages would cause CI failure. This was due to using an outdated Conda installation in the build environment, as referenced in issue #1672 .
I have updated the url for downloading Miniconda(3) & its PATH variable in the build environment to fix that issue, and now I can update the minimum supported package versions.
Thanks for the help @KamalakerDadi .

**Comment by GaelVaroquaux** [comment]
@kchawla-pi : is this ready for review and merge?

**Comment by GaelVaroquaux** [comment]
@bthirion, we're thinking the same :)

Release is getting closer. I would like to release an alpha in a few days, to give some exposure to the new interactive plotting function.

**Comment by kchawla-pi** [comment]
All the examples have been changed, and those require reviews, as do the tutorial in comments around them.
I have not changed all the documentation yet. I will do that tomorrow and then it should be ready for final review & merge.
I will also see if I can tone numpy==1.11.2 to numpy==1.11 .
Appveyor is failing, I don't have the privileges to trigger a rebuild. Will look into that tomorrow as well.

**Comment by kchawla-pi** [comment]
>  It requires bumping the version in nilearn/version, updating the docs, and the CI.

@GaelVaroquaux I assume by updating the docs you mean that in the documentation somewhere it is mentioned the minimum required versions of scipy, numpy, scikit-learn?
There are comments in the `.travis.yml` file regarding the build system's configs. They seem outdated. Should I remove them?
Specifically:
line 21    # Oldest supported versions (with neurodebian)
line 30    # Fake Ubuntu Xenial (Travis doesn't support Xenial yet)

**Comment by kchawla-pi** [comment]
The Appveyor build failed due to an HTTP resource not available error. Nothing to do with the code I think then.

**Comment by GaelVaroquaux** [comment]
> The Appveyor build failed due to an HTTP resource not available error. Nothing
> to do with the code I think then.

It is indeed unrelated.

I added an issue to tackle this problem:
https://github.com/nilearn/nilearn/issues/1705

**Comment by kchawla-pi** [comment]
Ready for review.

**Comment by kchawla-pi** [comment]
Ready for Review.

File: nilearn/decoding/searchlight.py [hunk 11]
```diff
@@ -166,10 +166,8 @@ def _group_iter_search_light(list_rows, estimator, X, y, groups,
     t0 = time.time()
     for i, row in enumerate(list_rows):
         kwargs = dict()
-        if not LooseVersion(sklearn.__version__) < LooseVersion('0.15'):
```
Comment (KamalakerDadi) [inline nilearn/decoding/searchlight.py]: Can you double check that this needs to be removed ?

File: nilearn/image/resampling.py [hunk 12]
```diff
@@ -520,20 +507,6 @@ def resample_img(img, target_affine=None, target_shape=None,
     else:
         resampled_data_dtype = data.dtype
 
-    if LooseVersion(scipy.__version__) >= LooseVersion('0.17'):
```
Comment (KamalakerDadi) [inline nilearn/image/resampling.py]: I am not sure this should be removed ?

**Comment by KamalakerDadi** [comment]
- matplotlib
  - examples//01_plotting/plot_3d_map_to_surface_projection.py needs to be updated because we no longer support matplotlib <= 1.3.1
  - LooseVersion check > 1.3.1 can be removed I think from nilearn/plotting/__init__.py
  - some clean ups in nilearn/_utils/fixes/matplotlib_backports.py

- numpy
  - Some clean ups in nilearn/_utils/extmath.py

- sklearn
   - some clean ups in nilearn/_utils/fixes/sklearn_validation.py
   - some clean ups in nilearn/_utils/__init__.py
   - Could you please check that ```check_X_y``` can be used from sklearn.utils directly than from sklearn_validation.py

- Bumping the older versions and supporting newer versions should be added in doc/whats_new.rst

- examples/03_connectivity/plot_group_level_connectivity.py is showing performance reduction in tangent space embedding when compared to partial correlation. @salma1601 can you please review this example ? Thanks

File: nilearn/decoding/searchlight.py (hunk 11 above)
Comment (kchawla-pi) [reply nilearn/decoding/searchlight.py]: It will always evaluate to False, now that SKLearn is minimum v0.18. The logic will never be executed.

File: nilearn/image/resampling.py (hunk 12 above)
Comment (kchawla-pi) [reply nilearn/image/resampling.py]: It will always evaluate to True, now that SciPy is minimum v0.17. The logic will always be executed. No need for a condition check.

**Comment by kchawla-pi** [comment]
> examples//01_plotting/plot_3d_map_to_surface_projection.py needs to be updated because we no longer support matplotlib <= 1.3.1

> Could you please check that check_X_y can be used from sklearn.utils directly than from sklearn_validation.py

Will check these out.

> nilearn/_utils/fixes/matplotlib_backports.py ; nilearn/_utils/extmath.py ; nilearn/_utils/fixes/sklearn_validation.py ; nilearn/_utils/init.py

The _utils module will be completely removed, so I am not making piecemeal changes to it.

> Bumping the older versions and supporting newer versions should be added in doc/whats_new.rst
Will do. It will be the last thing I do, after all the changes are finalized & working.

>  showing performance reduction in tangent space embedding when compared to partial correlation
I will ask you which tool etc you used to check this, so I will look for slowdowns & be mindful of such issues in future.

**Comment by kchawla-pi** [comment]
The CI is failing with some HTTP error, will look into this today.

**Comment by KamalakerDadi** [comment]
>The CI is failing with some HTTP error, will look into this today.

There seems like part of failures related to resampling.

File: examples/plot_decoding_tutorial.py [hunk 13]
```diff
@@ -215,16 +212,28 @@
...
+# and faces.
 session_label = behavioral['chunks'][condition_mask]
 
-from sklearn.cross_validation import LeaveOneLabelOut
-cv = LeaveOneLabelOut(session_label)
+# To leave a session out, we pass it to the groups parameter of cross_val_score.
+#
+# By default, cross_val_score uses a 3-fold KFold. We can control this by
+# passing the "cv" object, here a 5-fold:
 cv_score = cross_val_score(svc, fmri_masked, conditions, cv=cv)
```
[the remark this answers is not in the document]
Comment (kchawla-pi) [reply examples/plot_decoding_tutorial.py]: I believe in this case the example showcases only the use of 5-fold cv, without `groups` param; which also illustrates that the param is optional.
The comment `# To leave a session out...` refers to the next part, the one using LeaveOneGroupOut. I have moved the comment accordingly.
@GaelVaroquaux say if it isn't so.

File: nilearn/image/resampling.py (hunk 12 above)
Comment (kchawla-pi) [reply nilearn/image/resampling.py]: I had accidentally removed the logic as well, not just the condition check which was causing the problem. Fixed. Thanks.

File: examples/02_decoding/plot_haxby_grid_search.py [hunk 14]
```diff
@@ -97,8 +97,7 @@
 anova_svc.fit(X, y)
 y_pred = anova_svc.predict(X)
 
-from sklearn.cross_validation import LeaveOneLabelOut, cross_val_score
-cv = LeaveOneLabelOut(session[session < 10])
```
[the remark this answers is not in the document]
Comment (kchawla-pi) [reply examples/02_decoding/plot_haxby_grid_search.py]: It used to be:
`cv = LeaveOneLabelOut(session[session < 10])`

However, in subsequent code, the variable `cv` was not used anywhere. With `model_selection`, it becomes `cv = LeaveOneGroupOut()` which has no context sensitive data to initialize it, making its use more nebulous. None of the comments indicated why it was being used.
I removed `cv` to trace its effect but the examples & tests remained unaffected. 

It seemed to me that it might be a relic.
I can't fiddle with the examples directly yet. @GaelVaroquaux will have to change this part, including the explanatory comments, or tell me exactly what to write here.

File: examples/plot_decoding_tutorial.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (kchawla-pi) [reply examples/plot_decoding_tutorial.py]: Originally, there were 2 `cv = cross_vaL_score()` examples, one with `cv` parameter (now `groups`), and one without. This is the one without.

File: examples/plot_decoding_tutorial.py [hunk 15]
```diff
@@ -195,18 +197,13 @@
...
 print(cv_score)
 
 ###########################################################################
 # Note that we can speed things up to use all the CPUs of our computer
 # with the n_jobs parameter.
-#
-# By default, cross_val_score uses a 3-fold KFold. We can control this by
-# passing the "cv" object, here a 5-fold:
-cv_score = cross_val_score(svc, fmri_masked, conditions, cv=cv)
-print(cv_score)
```
[the remark this answers is not in the document]
Comment (kchawla-pi) [reply examples/plot_decoding_tutorial.py]: It has not been removed, a little rearranged.
@GaelVaroquaux You should take a look at this section `Line Number 177 to 232`, I have reworded the comments & moved a few lines of code and comments to reflect the new structure wrought by `model_selection`. I think I have done a reasonable job, but it would be best if you took a look at the entire section.

File: examples/plot_decoding_tutorial.py [hunk 16]
```diff
@@ -215,15 +212,25 @@
 #
 # The number of the session is stored in the CSV file giving the
 # behavioral data. We have to apply our session mask, to select only cats
-# and faces. To leave a session out, we pass it to a
-# LeaveOneLabelOut object:
+# and faces.
```
[the remark this answers is not in the document]
Comment (kchawla-pi) [reply examples/plot_decoding_tutorial.py]: cv = LeaveOneGroupOut() has been used twice. 
First time it was 
```
cross_val_score(cv=cv)
```
and then after the `#TO leave a session out...` comment:
```
cross_val_score(cv=cv, groups=session_label)
```
This is how I resolved it.
However, as I stated in the comment above, you should take a look at this section.

File: doc/conf.py [hunk 17]
```diff
@@ -22,7 +22,6 @@
 # is relative to the documentation root, use os.path.abspath to make it
 # absolute, like shown here.
 sys.path.insert(0, os.path.abspath('sphinxext'))
-import sphinx_gallery
```
[the remark this answers is not in the document]
Comment (kchawla-pi) [reply doc/conf.py]: It kept warning, `No Such Module as sphinx_gallery`. Usually I get the prompt to install the missing libraries into the env.
Adding it back.

**Comment by GaelVaroquaux** [comment]
The appveyor failure is spurious.

**Comment by kchawla-pi** [comment]
3/5 Travis Builds also failed. `HTTPError: HTTP Error 503: Service Temporarily Unavailable`
After changing one comment and one import. Local tests pass.
Don't think it's the code.
Will rerun Travis in a bit.

Meanwhile I will start studying the effects of removing the _utils module.

**Comment by GaelVaroquaux** [comment]
There are some violations of the coding style:
https://travis-ci.org/nilearn/nilearn/jobs/410277529

**Comment by kchawla-pi** [comment]
I will change TravisCI's Numpy to 1.11 instead of 1.11.2 After the next green builds, will push that.

**Comment by GaelVaroquaux** [comment]
> Meanwhile I will start studying the effects of removing the _utils module.

Just _utils.fixes, and maybe just one of the files in this directory.

File: examples/02_decoding/plot_haxby_grid_search.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (bthirion) [reply examples/02_decoding/plot_haxby_grid_search.py]: the 'cross_val_score' function in line 109 could receive an argument `groups=session[session < 10]`

**Comment by kchawla-pi** [comment]
I reverted the changes made to `plotting/__init__.py` as there was a merge conflict which I was ready to resolve. I will continue working on this in a new local branch.

**Comment by kchawla-pi** [comment]
@bthirion @GaelVaroquaux 
I am thinking about merging the work done so far in this PR (once the CI is working). We can push it to alpha release before the vacation.
The PR has achieved, by far and large, its stated purpose, to replace the use of cross_validation with model_selection in the examples.
I can submit another PR to remove any remaining references & code to older dependencies and the _utils.fixes module.

**Comment by kchawla-pi** [comment]
Have fixed plotting/__init__.py as indicated by Kamalaker in his code review.
Have removed nilearn._utils.fixes and all references to it.
Local tests pass.
The changes have been done in a separate branch, which I will rebase on to this PR or file as a new PR and push, once the existing code in this PR is greenlit.

File: nilearn/connectome/group_sparse_cov.py [hunk 18]
```diff
@@ -23,7 +23,7 @@
 from .._utils import CacheMixin
 from .._utils import logger
 from .._utils.extmath import is_spd
-from .._utils.fixes import check_cv
+from sklearn.model_selection import check_cv
```
Comment (KamalakerDadi) [inline nilearn/connectome/group_sparse_cov.py]: I would import where other sklearn imports are.

File: nilearn/decoding/space_net.py [hunk 19]
```diff
@@ -34,8 +31,9 @@
 from sklearn.metrics import accuracy_score
 from ..input_data.masker_validation import check_embedded_nifti_masker
 from .._utils.param_validation import _adjust_screening_percentile
-from .._utils.fixes import check_X_y
-from .._utils.fixes import check_cv, center_data
+from sklearn.utils import check_X_y
```
Comment (KamalakerDadi) [inline nilearn/decoding/space_net.py]: Same here: I would move imports to above where cluster of sklearn functions or classed are imported.

File: nilearn/decoding/searchlight.py [hunk 20]
```diff
@@ -26,7 +26,7 @@
 from ..image.resampling import coord_transform
 from ..input_data.nifti_spheres_masker import _apply_mask_and_get_affinity
 from .._utils.compat import _basestring
-from .._utils.fixes import cross_val_score
+from sklearn.model_selection import cross_val_score
```
Comment (KamalakerDadi) [inline nilearn/decoding/searchlight.py]: Same here: I would import sklearn above.

**Comment by KamalakerDadi** [comment]
There are some places which needs cleaning.

- sklearn
   - nilearn/connectome/group_sparse_cov.py -> check_cv
   - doc/decoding/decoding_intro.rst -> cross_val_score imports and imports in small demonstration blocks and leaveonelabelout.

File: nilearn/decoding/searchlight.py (hunk 11 above)
Comment (KamalakerDadi) [reply nilearn/decoding/searchlight.py]: >It will always evaluate to False

I could not understand how it would always evaluate to False.

**Comment by kchawla-pi** [comment]
@KamalakerDadi WHat does it to clean up at `nilearn/connectome/group_sparse_cov.py -> check_cv`?
@GaelVaroquaux Which file needs to e removed from `nilearn/_utils`?

**Comment by kchawla-pi** [comment]
One of the TravisCI builds is failing (the one with py==3.4) because in test_masking.test_compute_grey_matter(), there is code to assert that only 1 warning is generated and for some reason in this build, 2 warnings are generated.
I am guessing that bumping up the versions brought forth new warnings. It doesn't seem right that there be only one warning and no more, so long as the appropriate warning are generated.
Should I modify the code to assert_warns(expected warnings) and remove the assertion that number of warnings generated is 1?
Thanks for the help @jeromedockes

**Comment by kchawla-pi** [comment]
Is there a way to temporarily run only one test in Travis CI and not the rest?

**Comment by jeromedockes** [comment]
> Should I modify the code to assert_warns(expected warnings) and remove the assertion that number of warnings generated is 1?

I think so

**Comment by jeromedockes** [comment]
```
from nose.tools import assert_warns
# ...
 assert_warns(masking.MaskWarning, compute_background_mask, mean_image)
```

**Comment by kchawla-pi** [comment]
Giving this a shot.

**Comment by kchawla-pi** [comment]
BTW, The additional warning being generated was a RuntimeWarning by numpy, (incorrect value of median used) os something to that effect.

**Comment by jeromedockes** [comment]
sorry you need to import it like this
```
from nilearn._utils.testing import assert_warns
```

**Comment by KamalakerDadi** [comment]
>@KamalakerDadi What needs to cleaned up at nilearn/connectome/group_sparse_cov.py -> check_cv? I am not sure I understand.

l.951 # scikit-learn < 0.18

**Comment by KamalakerDadi** [comment]
That's a minor detail.

There are some import failures in few travis envs.

**Comment by GaelVaroquaux** [comment]
I've fixed the last errors and merged master in this.

If the CI is green, I'll merge tomorrow.

File: nilearn/decoding/searchlight.py (hunk 11 above)
Comment (kchawla-pi) [reply nilearn/decoding/searchlight.py]: @KamalakerDadi Have Fixed it. Take another look, compare it with the original code (not the one here).

**Comment by jeromedockes** [comment]
> I occassionally get this error while running local nose tests.
> FileNotFoundError: No such file or no access: '/tmp/nilearn_WYiXRj.nii'
> 
> When I rerun the test, it goes away.
> What's the solution?

could you tell us which test is failing like this?  I guess the problem comes from nilearn._utils.testing.write_tmp_imgs, as it creates a tempfile using mktemp with prefix="nilearn_" and suffix="nii", but this function is used in many places

**Comment by jeromedockes** [comment]
by the way, `mktemp` is used in many places in nilearn, it would be good to clean this up and replace it with `mkstemp` or `NamedTemporaryFile`

**Comment by kchawla-pi** [comment]
Will save the full traceback next time it happens and post it.

**Comment by GaelVaroquaux** [comment]
Everything ran!!!

Merging!!

Hurray!!

**Comment by kchawla-pi** [comment]
Haha wasn't just me sitting with bated breath. :-)

**Comment by kchawla-pi** [comment]
@GaelVaroquaux When do we officially close issues #1702 , #1672 , #1631 ?
